app/bundles/LeadBundle/Tests/Controller/Api/LeadApiControllerFunctionalTest.php

```diff
@@ -14,6 +20,20 @@ protected function setUp(): void
         // Disable API just for specific test.
         $this->configParams['api_enabled'] = 'testDisabledApi' !== $this->getName();
 
+        $this->useMockServices = false;
+
+        self::$container->set(
+            'session',
+            new Session(
+                new class() extends FixedMockFileSessionStorage {
+                    public function start()
+                    {
+                        Assert::fail('Session cannot be started during API call. It must be stateless.');
+                    }
+                }
+            )
+        );
+
```

Comment:
I tested this approach locally, but it doesn't work.
Tests will fail as you can't change an already defined service.

I have another approach that I would suggest, I'll create a separate PR, as It's a multi file change

Reply:
@mollux I'm removing random PRs from the 5.0.0-alpha milestone and looking at why the PRs are stuck. Do you have some update on this?